### ISSUE TITLE
Allow user to specify which script a subcommand should call

### ISSUE THREAD

**Issue by matthewadams** [tj/commander.js#532]
If the script a user wants to execute as a subcommand doesn't abide by the naming convention that's documented (like [`pm install` looks for script `pm-install`](https://github.com/tj/commander.js/#git-style-sub-commands)), the user should be able to specify the script that's to be executed.   For example:

```
// file: ./examples/pm
var program = require('..');

program
  .version('0.0.1')
  .command('install [name]', 'install one or more packages', 'myinstall') // (1)
  .command('search [query]', 'search with optional query', '../others/q.js') // (2)
  .command('list', 'list packages installed', {isDefault: true}, '/home/me/foo/bar/mylist.js') // (3)
  .parse(process.argv);
```

In `(1)`, commander would assume the same directory as the current file, look for `./myinstall`, then if not found,  append `.js`, resulting in `./myinstall.js`.
In `(2)`, commander would use the path relative to the current file.  No `.js` appended since there's already a `.js` extension.
in `(3)`, commander would use the absolute path.  No `.js` appended since there's already a `.js` extension.

**Comment by b4dnewz** [comment]
2 years... will this issue never be addressed? :disappointed: it is an extremely useful feature that you can use an alternative naming convention or specify a file

**Comment by abetomo** [comment]
Thanks for your suggestion.
I will add features.

**Comment by shadowspawn** [comment]
Pull Request #532 adds an option for this.

I am interested in feedback on `executableFile` as the option name?

e.g.
```
program
.command("example", "example description", { executableFile: "myExampleCommand" });
```

**Comment by shadowspawn** [comment]
Pull Request being considered for v3 is #999

**Comment by shadowspawn** [comment]
This issue will be resolved when v3.0.0 is released. Available now as a prerelease. See #1001

**Comment by shadowspawn** [comment]
`.command` now supports the `executableFile` option (thanks to @abetomo). 
Shipped in v3: https://github.com/tj/commander.js/releases/tag/v3.0.0
